src/parser/sparqlParser/SparqlQleverVisitor.cpp

```diff
@@ -389,6 +554,32 @@ ParsedQuery Visitor::visitTypesafe(Parser::SelectQueryContext* ctx) {
                        ctx->getText());
 }
 
+// ____________________________________________________________________________________
+GraphPatternOperation::Subquery Visitor::visitTypesafe(
+    Parser::SubSelectContext* ctx) {
+  ParsedQuery query;
+  visibleVariables_.emplace_back();
+  query._clause = visitTypesafe(ctx->selectClause());
+  query._rootGraphPattern = visitTypesafe(ctx->whereClause());
+  query.setNumInternalVariables(numInternalVariables_);
+  query.addSolutionModifiers(visitTypesafe(ctx->solutionModifier()));
+  numInternalVariables_ = query.getNumInternalVariables();
+  auto values = visitTypesafe(ctx->valuesClause());
+  if (values.has_value()) {
+    query._rootGraphPattern._children.emplace_back(std::move(values.value()));
+  }
+  // Process Variables that are visible by children of this query.
+  for (const auto& variable : visibleVariables_.back()) {
+    query.registerVariableVisibleInQueryBody(variable);
+  }
+  visibleVariables_.pop_back();
+  // Variables that are selected in this query are visible in the parent query.
+  for (const auto& variable : query.selectClause().getSelectedVariables()) {
+    visibleVariables_.back().emplace_back(variable);
+  }
+  return {query};
```

Comment:
are the braces needed?

Reply:
Ah, probably the constructor is explicit. Let me check again (and check yourself) if we need to write `std::move` in this case.